src/unstract/adapters/vectordb/pinecone/src/static/json_schema.json

```diff
@@ -19,12 +19,96 @@
       "format": "password",
       "description": "Provide the API key for your Pinecone account"
     },
-    "environment": {
+    "spec": {
       "type": "string",
-      "title": "Environment",
-      "type": "string",
-      "default": "us-west1-gcp",
-      "description": "Provide the environment for your Pinecone account. Example us-west1-gcp-free"
+      "title": "Specification",
+      "enum": [
+        "pod",
+        "serverless"
+      ],
+      "default": "serverless",
+      "description": "Configurations describing how the index should be deployed"
+    }
+  },
+  "allOf": [
+    {
+      "if": {
+        "properties": {
+          "spec": {
+            "const": "pod"
+          }
+        }
+      },
+      "then": {
+        "properties": {
+          "environment": {
+            "type": "string",
+            "title": "Environment",
+            "default": "us-west1-gcp",
+            "description": "Provide the environment for your Pinecone account. Example us-west1-gcp-free"
+          },
+          "replicas": {
+            "type": "string",
+            "title": "Number of replicas",
+            "default": "1",
+            "description": "The number of replicas to deploy for the pod index.",
```

Comment:
NIT: @gaya3-zipstack we could specify that the user can control these through their Pinecone console / UI directly for all the readOnly attributes we have marked

Reply:
Yea... Will pick it up with the next set of adapter changes! @chandrasekharan-zipstack